docs/develop/func/cookbook.md

```diff
@@ -730,11 +786,12 @@ throw(36); ;; the error will be triggered anyway
 
 ### Reversing tuples
 
-TODO: please provide some description like in a `Modulo operations` section
+Because tuple stores data as a stack, sometimes we have to reverse tuple to read data from the other end.
 
 ```func
 forall X -> (tuple, X) ~tpop (tuple t) asm "TPOP";
 int tuple_length (tuple t) asm "TLEN";
+forall X -> (tuple) to_tuple (X x) asm "NOP";
 
 (tuple) reverse_tuple (tuple t1) {
```

Comment:
This function can be optimized using EXPLODEVAR instruction.
This could be too much for docs, but I think it needs to be stated.